### PR TITLE
Add ExpirationManager and specs

### PR DESCRIPTION
**Pull request by dudoslav**
ExpirationManager deletes invalid expirations and sets default ones

### CODE REVIEW

File: lib/berta/expiration_manager.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+module Berta
+  # Class for managing expiration dates on vms
+  class ExpirationManager
+    # Update all expirations on vm, removes invalid ex[irations
```
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: Typo ex[irations -> expirations

File: lib/berta/expiration_manager.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+      vms.each { |vm| add_default_expiration(vm) }
+    end
+
+    # Removes invalid expirations on vm
+    #
+    # @param [VirtualMachineHandler] vm
+    def remove_invalid_expirations(vm)
+      exps = vm.expirations
+      exps.keep_if \
+        { |exp| exp.time.to_i <= Time.now.to_i + Berta::Settings.expiration_offset }
```
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: Shouldn't this be `exp.time.to_i > Time.now.to_i + Berta::Settings.expiration_offset`? What expirations are you trying to remove?

File: lib/berta/expiration_manager.rb [hunk 3]
```diff
@@ -0,0 +1,33 @@
+module Berta
+  # Class for managing expiration dates on vms
+  class ExpirationManager
+    # Update all expirations on vm, removes invalid ex[irations
+    #   and if needed will set default expiration date
+    #
+    # @param [Array<VirtualMachineHandler>] virtual machines
+    #   to update expiration on
+    def update_expirations(vms)
```
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: Can't this be done in one `each` cycle?

File: lib/berta/expiration_manager.rb (hunk 2 above)
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: Statement `Time.now.to_i + Berta::Settings.expiration_offset` is used multiple times in the code. Maybe you can set an attribute to this value in costructor and use that instead.

Comment (dudoslav) [inline lib/berta/expiration_manager.rb]: Im keeping all expirations that are inside expiration interval. That means that if expiration date is further away from now than default expiration offset i shouldn't keep it.

File: lib/berta/expiration_manager.rb (hunk 3 above)
Comment (dudoslav) [inline lib/berta/expiration_manager.rb]: Somehow unit tests wouldn't pass in only one cycle

File: lib/berta/expiration_manager.rb (hunk 2 above)
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: Sure, ok, my mistake.

File: lib/berta/expiration_manager.rb (hunk 3 above)
Comment (Misenko) [inline lib/berta/expiration_manager.rb]: 😄 Well that's odd. Try to figure out why.

Comment (dudoslav) [inline lib/berta/expiration_manager.rb]: I have a clue, when it was in one each cycle the vm.expirations returned same number of expirations even if remove_invalid_expirations removed some expirations. Now the problem is that I dont understand why it works in in separate each cycles. But I will look into it.

Comment (Misenko) [inline lib/berta/expiration_manager.rb]: That's what I was wondering. I guess you should call `vm.info` on a `OpenNebula::VirtualMachine` instance before each data retrieval you do. I know it's annoying but that instance is not updated automatically to reflect the changes on the server.

File: spec/berta/virtual_machine_handler_spec.rb [hunk 4]
```diff
@@ -99,7 +99,9 @@
     context 'with empty array', :vcr do
       it 'does nothing' do
         service.running_vms.each do |vm|
+          l = vm.expirations.length
```
Comment (Misenko) [inline spec/berta/virtual_machine_handler_spec.rb]: What the hell does this do? :flushed:

Comment (dudoslav) [inline spec/berta/virtual_machine_handler_spec.rb]: Nothing :D

Comment (dudoslav) [inline spec/berta/virtual_machine_handler_spec.rb]: just kidding, it tests if method update_expiration wont change anything when called with empty array as parameter

Comment (Misenko) [inline spec/berta/virtual_machine_handler_spec.rb]: Sorry. Stupid font. I thought its a number '1' not a letter 'l'...
Anyway. Don't use one letter variable names unless it's a mathematical/physical variable.

**Pull request merged**